core/src/main/java/org/axonframework/eventsourcing/eventstore/jdbc/JdbcEventStorageEngine.java

```diff
@@ -413,7 +413,17 @@ protected PreparedStatement readSnapshotData(Connection connection, String ident
      * @throws SQLException when an exception occurs while creating the event data
      */
     protected DomainEventData<?> getDomainEventData(ResultSet resultSet) throws SQLException {
-        return (DomainEventData<?>) getTrackedEventData(resultSet, null);
+        long globalSequence = resultSet.getLong(schema.globalIndexColumn());
+        TrackingToken trackingToken = new GlobalSequenceTrackingToken(globalSequence);
```

Comment:
Actually it won't. As far as I could see the only code path that leads to `protected DomainEventData<?> getDomainEventData(ResultSet resultSet)` being called is when an aggregate is loaded from the repository based on an aggregate identifier.

When loading events from an EventProcessor only `protected DomainEventData<?> getTrackedEventData(ResultSet resultSet) ` is being called.